### PR TITLE
Test Proxy Space Lister

### PR DESCRIPTION
**Pull request by rajivnathan**
Tests the proxy space lister API

Paired with https://github.com/codeready-toolchain/registration-service/pull/301

### CODE REVIEW

File: test/e2e/proxy_test.go [hunk 1]
```diff
@@ -231,14 +233,114 @@ func TestProxyFlow(t *testing.T) {
...
+				)
+
+				t.Run("successful list", func(t *testing.T) {
+					// when
+					workspaces := &toolchainv1alpha1.WorkspaceList{}
+					err = proxyCl.List(context.TODO(), workspaces)
+
+					// then
+					require.NoError(t, err)
+					require.Len(t, workspaces.Items, 1)
```
Comment (alexeykazakov) [inline test/e2e/proxy_test.go:267]: sWe need to cover the case when there is actually some workspace sharing and the list of available workspaces contains more than one workspace.

Let's make sure we test a real life scenarios.
Create multiple spaces & spacebindings. One user has access to a single workspace only. And at least two more users which share their spaces with each other. Then test that:
- each user gets expected list of spaces
- doesn't get the workspaces which the user doesn't have access to included into the result (so no noise workspaces included)
- same for GET specific workspace request.

Comment (rajivnathan) [reply test/e2e/proxy_test.go]: @alexeykazakov e2e tests have been updated

**Comment by alexeykazakov** [comment]
/retest

after merging https://github.com/codeready-toolchain/member-operator/pull/408

**Comment by rajivnathan** [comment]
/retest
after merging https://github.com/codeready-toolchain/host-operator/pull/742

**Pull request merged**